odoo/addons/base/models/ir_model.py

```diff
@@ -470,10 +470,12 @@ def _related_field(self):
                 raise UserError(_("Unknown field name '%s' in related field '%s'") % (name, self.related))
             if index < last and not field.relational:
                 raise UserError(_("Non-relational field name '%s' in related field '%s'") % (name, self.related))
+            if self.store and index < last and not field._description_searchable:
+                raise UserError(_("Field %s (%s) cannot be used in stored related field.") % (field.string, field.name))
```

Comment:
@rco-odoo I'd prefer to keep the check for `self.store` to minimize changes in stable. We may apply broader restrictions in master